elk/utils/data_utils.py

```diff
@@ -120,3 +120,15 @@ def binarize(template: Template, label: int, new_label: int, rng: Random) -> Tem
     )
 
     return new_template
+
+
+def is_streamable(ds: IterableDataset, max_examples: int) -> bool:
```

Comment:
oh is this actually fast? I guess this could be okay... I was imagining something where like, `BalancedSampler` internally checks if it's skipped over too many samples without yielding anything, maybe throwing `StopIteration` or whatever in that case, and then we pick up on that

But idk maybe this is just too complex